internal/secretless.go

```diff
@@ -0,0 +1,39 @@
+package internal
+
+import (
+	"log"
+
+	v1 "github.com/cyberark/secretless-broker/internal/plugin/v1"
+	"github.com/cyberark/secretless-broker/pkg/secretless"
+	v2 "github.com/cyberark/secretless-broker/pkg/secretless/config/v2"
+)
+
+
+// TODO: move to impl package
+type _secretless struct {
+	config v2.Config
+	logger log.Logger
+	eventNotifier v1.EventNotifier
+	proxyServices []ProxyService
+}
+
+func (s *_secretless) Start() {
+	// TODO: start ProxyServices, etc
+}
+
+func (s *_secretless) Stop() {
+	// TODO:
+}
+
+// will be constructed in main
+func NewSecretless(cfg v2.Config, logger log.Logger, evtNotifier v1.EventNotifier) secretless.Secretless {
+	ret := _secretless{
+		config: cfg,
+		logger: logger,
+		eventNotifier: evtNotifier,
+	}
+
+	// TODO: create our unstarted ProxyServices here (sync with Kumbi)
+
+	return &ret
```

Comment:
`ret` doesn't really tell us what type/object we're working with

Reply:
Raised in channel for resolution as code convention

Reply:
So let's do this, I'll go ahead and change it to something descriptive for now, and then I'll add the question to best practices doc which we'll eventually share in #about-golang, and we can get feedback on various decisions like this and perhaps vote on them, and eventually we'll have a nice best practices doc which us and the IL devs can all share.